### PR TITLE
Switch to using the cypher query builder for the createUser service

### LINKED ISSUE #200: CreateUser optional properties incorrectly required by DB query
`phone`, `timezone`, `bio` are optional properties of create user.
The DB query should not require them to be set.

### PR DESCRIPTION
**Pull request by richardgirges**
Closes #200 

In this PR:
* UserService.create switched to utilize the cypher query builder for the DB insertion
* Removed some debug code from the DB service that got merged in a prior PR

I attempted to use the `db.createNode` generic. This turned out to be a pretty big waste of time that required too much refactoring. Due to the access control props we're setting on the user's base node, I think it justifies utilizing the query builder. Perhaps we can revisit this later if the `db.createNode` proves to be less flexible in other services.

### CODE REVIEW

File: src/components/user/user.service.ts [hunk 1]
```diff
@@ -209,58 +207,131 @@ export class UserService {
...
+          relation('out', '', 'email', {
+            active: true,
+            createdAt: 'datetime()',
+          }),
+          node('email', 'EmailAddress:Property', {
             active: true,
-            value: $email,
-            createdAt: datetime()
+            value: input.email,
+            createdAt: 'datetime()',
```
Comment (CarsonF) [inline src/components/user/user.service.ts]: This is interpreted as a literal string with the query builder.
```suggestion
            createdAt: DateTime.local().toNeo4JDateTime(),
```
Feel free to make variable to not repeat yourself.

File: src/components/user/user.service.ts [hunk 2]
```diff
@@ -209,58 +207,131 @@ export class UserService {
...
+          node('user'),
+          relation('out', '', 'displayLastName', {
+            active: true,
+            createdAt: 'datetime()',
+          }),
+          node('displayLastName', 'Property', {
+            active: true,
+            value: input.displayLastName,
+          }),
+        ],
```
Comment (CarsonF) [inline src/components/user/user.service.ts]: Most of these property queries are the same. How about a helper function?
```ts
const property = (prop: string, value: any) => {
  if (!value) {
    return [];
  }
  return [
    node('user'),
    relation('out', '', prop, {
      active: true,
      createdAt: DateTime.local().toNeo4JDateTime(),
    }),
    node(prop, 'Property', {
      active: true,
      value,
    }),
  ];
};

.create([
  ...
  property('phone', input.phone),
  ...
])
```

Comment (richardgirges) [reply src/components/user/user.service.ts]: Love this idea. Should this be an enclosed method you think? Or a private method in the db service class?

File: src/components/user/user.service.ts (hunk 1 above)
Comment (richardgirges) [reply src/components/user/user.service.ts]: Changes applied!

File: src/components/user/user.service.ts (hunk 2 above)
Comment (richardgirges) [reply src/components/user/user.service.ts]: Changes applied!

File: src/components/user/user.service.ts [hunk 3]
```diff
@@ -227,22 +229,39 @@ export class UserService {
...
+    const property = (prop: string, value: any) => {
+      if (!value) {
+        return [];
+      }
+
+      return [
+        node('user'),
+        relation('out', '', prop, {
+          active: true,
+          createdAt,
```
Comment (CarsonF) [inline src/components/user/user.service.ts]: This can reference the variable defined below?

Comment (richardgirges) [reply src/components/user/user.service.ts]: const vars get hoisted. But I can see how that looks a little weird, I'll move it to the top

Comment (richardgirges) [reply src/components/user/user.service.ts]: Okay! latest updates pushed!

Comment (CarsonF) [reply src/components/user/user.service.ts]: Yeah you're right. But I agree it looks better now, thanks!

**Pull request merged**